Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -361,6 +380,154 @@ public sealed class Server : ServerBase, IDisposable {
             return item;
         }
 
+        public override async Task<SignatureHelp> SignatureHelp(TextDocumentPositionParams @params) {
+            var uri = @params.textDocument.uri;
+            GetAnalysis(@params.textDocument, @params.position, @params._version, out var entry, out var tree);
+
+            if (_traceLogging) {
```

Comment:
If we have `MessageType` parameter that controls severity, why do you need to check for `_traceLogging`?

Reply:
The parameter doesn't decide whether the event gets raised or not, only what the client does with it once it has received it. Trace logging is for extremely verbose messages that we don't even want to send in most cases, or they'll clog up the IPC channel and hurt performance (particularly in remote scenarios).

Reply:
Ok, then how about introducing a helper method for it:
```cs
private void LogIfTraceEnabled(string message) {
    if (_traceLogging) {
        LogMessage(MessageType.Log, message);
    }
}
```

Reply:
Sure. I can do that